src/theory/datatypes/sygus_datatype_utils.h

```diff
@@ -103,8 +103,11 @@ Node mkSygusTerm(const DType& dt,
 /**
  * Same as above, but we already have the sygus operator op. The above method
  * is syntax sugar for calling this method on dt[i].getSygusOp().
+ *
+ * Note that this method should not be used on sygus operators that denote
+ * the "any constant" constructor.
```

Comment:
What's wrong with the old behavior?

Reply:
The special case for constructing builtin terms for the any-constant constructor has been moved to the method where we have access to the datatype constructor.